tests/test_dataset.py

```diff
@@ -0,0 +1,253 @@
+    assert dataset.name_link == expected_name_link
+
+    # publication: skipped for now
+    # repository_link : skipped for now
+
+    # sampling_rates
+    expected_sampling_rates = ', '.join(set([
+        str(audiofile.sampling_rate(file)) for file in db.files
+    ]))
+    assert dataset.sampling_rates == expected_sampling_rates
+
+    # schemes
+    expected_schemes = audbcards.core.dataset.format_schemes(db.schemes)
+    assert dataset.schemes == expected_schemes
+
+    # short_description
+    max_desc_length = 150
+    expected_description = db.description if (
+        len(db.description) < max_desc_length
+    ) else f'{db.description[:max_desc_length - 3]}...'
+    assert dataset.short_description == expected_description
+
+    # version_link
+    github = 'https://github.com/audeering'
+    branch = 'main'
+    url = f'{github}/{db.name}/blob/{branch}/CHANGELOG.md'
+    expected_version_link = f'`{pytest.VERSION} <{url}>`__'
+    assert dataset.version_link == expected_version_link
+
+    # Test player
+    # Init player using dataset object
+    plot_waveform = True
+    player_str = dataset.player(dataset.example, waveform=plot_waveform)
+    # Generate expected player
+    expected_player_str = ''
+    # Check if file has been copied under the build folder
+    dst_dir = f'{BUILD}/datasets/{db.name}'
+    assert os.path.exists(os.path.join(dst_dir, expected_example))
+    # Add plot of waveform
+    if plot_waveform:
+        signal, sampling_rate = audiofile.read(
+            os.path.join(dst_dir, expected_example),
+            always_2d=True,
+        )
+        plt.figure(figsize=[3, .5])
+        ax = plt.subplot(111)
+        audplot.waveform(signal[0, :], ax=ax)
+        audbcards.core.dataset.set_plot_margins()
+        plt.savefig(f'{os.path.join(tmpdir, db.name)}.png')
+        plt.close()
+        # Check if generated images are exactly the same (pixel-wise)
+        assert open(f'{db.name}.png', 'rb').read(
+        ) == open(f'{os.path.join(tmpdir, db.name)}.png', 'rb').read()
+        # Append image to the expected player_str
+        expected_player_str += (
+            f'.. image:: ../{db.name}.png\n'
+            '\n'
+        )
+    # Append audio to the expected player_str
+    expected_player_str += (
+        '.. raw:: html\n'
+        '\n'
+        f'    <p><audio controls src="{db.name}/{expected_example}">'
+        f'</audio></p>'
+    )
+    # Check if the generated player_str and the expected matches
+    assert expected_player_str == player_str
+
+
+@pytest.mark.parametrize(
+    'scheme_names, scheme_dtypes, labels',
+    [
+        (
+            ['emotion', 'age', 'gender', 'language', 'speaker'],
+            ['str', 'int', 'str', 'str', 'int'],
+            [
+                ['happy', 'sad'],
+                None,
+                ['female', 'male'],
+                ['DE', 'EN'],
+                'speaker',
+            ],
+        ),
```

Comment:
It is very hard to understand this test, maybe it is easier to provide the expected result as argument and test only a few combinations?

Reply:
I've changed it accordingly and added two cases for this test

Reply:
Thanks.